src/log.c

```diff
@@ -68,8 +68,15 @@ log_open(struct view *view, enum open_flags flags)
 			"--stat", use_mailmap_arg(), "%(logargs)", "%(cmdlineargs)",
 			"%(revargs)", "--no-color", "--", "%(fileargs)", NULL
 	};
+	enum status_code code;
 
-	return begin_update(view, NULL, log_argv, flags);
+	code = begin_update(view, NULL, log_argv, flags);
+	if (code != SUCCESS)
+		return code;
+
+	watch_register(&view->watch, WATCH_HEAD | WATCH_REFS);
```

Comment:
Changes look good, I just have a related question - I can't figure out how to make `set refresh-mode = after-command` work. It doesn't refresh after I run `:!echo` and type `q`.

Reply:
I must say that I don't use it either. Wondering if anyone does... WATCH_EVENT_AFTER_COMMAND seems to be trigged only after opening an editor or running a command requiring confirmation.

Reply:
Hehm, yeah, the refresh code is really buggy unfortunately. It probably needs to be completely redesigned. Might be better to rip it out.